src/Agent.Listener/Configuration/WindowsSecurityManager.cs

```diff
@@ -0,0 +1,477 @@
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Security.AccessControl;
+using System.Security.Cryptography;
+using System.Security.Principal;
+using System.Text;
+
+using Microsoft.VisualStudio.Services.Agent.Util;
+
+namespace Microsoft.VisualStudio.Services.Agent.Listener.Configuration
+{
+    [ServiceLocator (Default = typeof(WindowsSecurityManager))]
+    public interface IWindowsSecurityManager : IAgentService
+    {
+        string GetUniqueBuildGroupName();
+
+        bool LocalGroupExists(string groupName);
+
+        void CreateLocalGroup(string groupName);
+
+        void AddMemberToLocalGroup(string accountName, string groupName);
+
+        void GrantFullControlToGroup(string path, string groupName);
+
+        bool GrantUserLogonAsServicePrivilage(string accountName);
+
+        bool IsValidCredential(string logonAccount, string logonPassword);
+
+        NTAccount GetDefaultServiceAccount();
+
+        void SetPermissionForAccount(string path, string accountName);
+    }
+
+    public class WindowsSecurityManager : AgentService, IWindowsSecurityManager
+    {
+        private const string AgentServiceLocalGroupPrefix = "VSTS_AgentService_G";
```

Comment:
This is different from the current group name. Have you thought about when the .Net Core agent slips in as an update for the current Windows agent? For example, on uninstall will it remove the old group?

Don't get me wrong, it doesn't need to be completely thought through yet. But probably should add some TODOs in the code wherever more thought is required w.r.t. that problem.

Until it is thought through, consider keeping the old name.